zad2/warriors.rb

```diff
@@ -1,12 +1,89 @@
-class Warrior
-  def initialize(name:, level:)
-  end
+class Character
+	attr_reader :name
+	attr_accessor :level
 
-  def strength
-  end
+	def initialize(name, level) 
+		@name = (name.is_a? String) ? name : "NoName Hero"
+		@level = ((level.is_a? Integer) && (level < 100)) ? level : 1
+	end
+
+	def card
+		p "#{name} (lvl #{level})"
+	end
+
+	def strength
+  		level + [*1..12].sample
+  	end
+end
+
+class Warrior < Character
+  	def initialize(name, level)
+  		super(name, level)
+  	end  	
```

Comment:
nie ma potrzeby pisania tej metody. Jeśli ruby nie znajdzie `initialize` w klasie `Warrior` to automatycznie zacznie go szukać w klasach, z których dziedziczy. 😃 nadpisanie byłoby użyteczne gdybyśmy, oprócz `initialize` z `Character`, chcieli zrobić coś co jest unikalne dla klasy `Warrior`